_posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md

```diff
@@ -0,0 +1,109 @@
+
+so you have more control on the call and it gets easier to do batch operations.
+We also fixed some bugs and updated some dependencies.
+
+Testing things in the console was fun, but we needed a real life scenario to check that the gem was working correctly and to find missing features.
+Lucky for us, our Product Owner came up with task perfect for this situation.
+We needed to create a Sales Analytics App for the ePages [App Store](https://blog.epages.com/us/2016/05/14/the-epages-app-store-everything-for-your-e-commerce-success/).
+
+## The new app in Rails
+
+One of the things that I like the most about working at ePages is that we have lots of colleagues willing to help us.
+We had mockups, designs and texts for the whole Web App, so we could focus on programming.
+
+For simplicity we mimicked more or less the API models and added a bit of Rails Magic so it's even easier to use.
+
+{% image blog/blog-sales-app-model.png auto center %}
+
+After that, we created all the sections of the app: Dashboard, Sales, Products, and Customers.
+All these parts are bound to two dates (start date and end date) so the merchant can get even more useful data faster.
+
+{% imagebasic blog/blog-sales-app-dashboard.png %} image {% endimagebasic %}
+
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+{% imagebasic blog/blog-sales-app-sales.png %} image {% endimagebasic %}
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+{% imagebasic blog/blog-sales-app-customers.png %} image {% endimagebasic %}
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
+Above the table we can see all the customers for the timeframe and if we had any returning ones.
+
+{% imagebasic blog/blog-sales-app-products.png %} image {% endimagebasic %}
+
+The last part are the Products.
+In this view, we have the products ordered via revenue or amount of units sold.
+Here we implemented the functionality of having more insights on the product.
+When clicking on one, you can see the extra data or cross selling products, this is, what other items have shared orders with.
+This allows the merchant to maybe realize on connections (or disconnections) between products in an easy and fast way.
+
+Both, the Customers and Products section, have the option to download the data on a `csv`.
+Other basic functionalities like changing you email address or password and 'Forgot my password' are also included in the app.
```

Comment:
Other basic functionalities like changing the email address or password as well as a 'Forgot my password' option are also included in the app.